kadmin/server.c

```diff
@@ -139,8 +139,12 @@ kadmind_dispatch(void *kadm_handlep, krb5_boolean initial,
     }
     case kadm_delete:{
 	op = "DELETE";
+	if (readonly) {
```

Comment:
Pull this out and do a switch above.  That will make it much easier to review, reason about, maintain.